os/kernel/binary_manager/binary_manager_load.c

```diff
@@ -420,31 +435,38 @@ static int binary_manager_reload(char *bin_name)
 static int loading_thread(int argc, char *argv[])
 {
 	int ret;
-	int type;
+	int load_cmd;
+	int reload_type;
 
 	if (argc <= 1) {
 		bmdbg("Invalid arguments\n");
 		return -1;
 	}
 
 	/* Arguments : [1] type */
-	type = (int)atoi(argv[1]);
+	load_cmd = (int)atoi(argv[1]);
 
 	ret = BINMGR_INVALID_PARAM;
-	switch (type) {
+	switch (load_cmd) {
 	case LOADCMD_LOAD_ALL:
 		ret = binary_manager_load_all();
 		break;
 	case LOADCMD_RELOAD:
+	case LOADCMD_UPDATE:
 		if (argc <= 2) {
 			bmdbg("Invalid arguments for reloading, argc %d\n", argc);
 			break;
 		}
+		if (load_cmd == LOADCMD_RELOAD) {
+			reload_type = RELOAD_FAULT;
```

Comment:
I don't understand why RELOAD_FAULT?
FAULT but it will load binary through the binary_manager_load_binary.....
And command is just 'LOADCMD_RELOAD' but version is latest, it ignore version and reload again?
I mean only update checks the version?

Reply:
In no update situations, reload is only a fault condition. So in that case you have to reload unconditionally.

Reply:
That's what I am saying now, you are not making error detector, it is loader, right?
Detecting Fault is not role of this file, it is just requested 'LOAD' from detector.
So fault is not good name of command.
See below.
```
thread Detector()
{
   while(true) {
    if (fault_condition) {
           load_binary(LOAD);
    } else if (version_is_updated) {
           load_binary(UPDATE);
    }
   }
}

void load_binary(int cmd)
{
    if (cmd == LOAD) {
            load_binary();
    } else if (cmd == UPDATE) {
            update_binary();
    }
}
}